### ISSUE TITLE
How to get device token when using VOIP push?

### ISSUE THREAD

**Issue by BorntraegerMarc** [phonegap/phonegap-plugin-push#2442]
### Expected Behaviour
`pushObject.on('registration')` returns the device token when using VOIP push.

### Actual Behaviour
The callback is never called.

### Reproduce Scenario (including but not limited to)
```
const options: any = {
                    ios: {
                        voip: true
                    }
                };
                const pushObject: PushObject = this.push.init(options);
```
#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
iOS 12

#### Cordova CLI version and cordova platform version

    cordova --version                                    # e.g. 6.0.0
    cordova platform version android                     # e.g. 4.1.1

#### Plugin version

    cordova plugin version | grep phonegap-plugin-push   # e.g. 1.5.3

**Comment by macdonst** [comment]
@BorntraegerMarc we haven't done any work on iOS 12. Support will be in a future release.

**Comment by BorntraegerMarc** [comment]
It's not an issue with iOS 12 I believe. Also get the same behavior with iOS 11. Should the tokens be passed in the callback of "registration" event?

**Comment by BorntraegerMarc** [comment]
Because this is the only output I'm seeing from my terminal:
```
2018-06-26 16:18:59.961830+0200 Komed Health[418:26737] subscribe from topic: my-topic
2018-06-26 16:19:00.011608+0200 Komed Health[418:26737] Successfully subscribe to topic my-topic
2018-06-26 16:19:00.011685+0200 Komed Health[418:26737] THREAD WARNING: ['PushNotification'] took '49.866211' ms. Plugin should use a background thread.
2018-06-26 16:19:00.011819+0200 Komed Health[418:26846] 4.13.0 - [Firebase/Messaging][I-FCM002010] Cannot subscribe to topic: my-topic with token: (null)
2018-06-26 16:19:00.012499+0200 Komed Health[418:26846] Push Plugin VoIP set to true
2018-06-26 16:19:00.014723+0200 Komed Health[418:26737] XXX success
```

**Comment by macdonst** [comment]
@BorntraegerMarc wait are you trying to do VoIP and FCM at the same time?

**Comment by macdonst** [comment]
@BorntraegerMarc I'm just realizing you are using typescript and I don't think the `voip` option is in our types definition.

**Comment by fredgalvao** [comment]
@macdonst Even though you're right on that, it would cause a totally different issue unrelated to `registration` not being triggered.

**Comment by BorntraegerMarc** [comment]
Yeah. I'm not typing the init function so the normal JS method is executed. (But the types should be updated ;) )
